src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs

```diff
@@ -278,6 +283,11 @@ internal unsafe QuicConnection(QUIC_HANDLE* handle, QUIC_NEW_CONNECTION_INFO* in
             throw;
         }
 
+        if (NetEventSource.Log.IsEnabled())
+        {
+            NetEventSource.Info(this, $"{this} New server connection.");
```

Comment:
Nit: I'd go with "inbound"-"outbound" and/or "accepted"-"opened" terminology (which is used in the API shape as well), rather than using "client"-"server" terminology, which also technically might sound a bit ambiguous ("server connection" can be read both as "on the server-side" and "to a server")